### PR TITLE
Change Meteors to use Universal State code

### PR DESCRIPTION
**Pull request by ghost**
There shouldn't have been any gameplay change, this is solely a refactor
- Meteor Storm Universal State now handles all the meteor storm stuff, like calling the shuttle, spawning supplies and spawning meteor waves
- Meteor gamemode is still around but has been trimmed to only handle delay to announcement, at which point it passes the task to the universal state handler
- Added a proc that handles passing variables to the meteor's universal state specifically
- Added a buzzer sound (warning.ogg, file already existed but wasn't used) just before announcement for ambiance and what not
- Added a new verb under Secrets that triggers the Meteor Storm universal state (just like Nar'Sie and Supermatter)
- Tweaked the end game results

Due to the shuttle controller (the one that handles shuttle calls, not the brand new shuttle controller we added a few months ago) being complete spaghetti, the meteor storm event does not handle the shuttle well. AKA, warning was given to not fire this if the shuttle has been called. Otherwise, it doesn't apply the proper time or does other very weird stuff. Up to the admins to not fuck up, I have given proper written warning in the actual verb

Should not affect meteor gamemode since the shuttle is called before shuttle calls are allowed

### CODE REVIEW

**Comment by 9600bauds** [comment]
Seems like this would conflict with #10352 which has merge priority

**Comment by PJB3005** [comment]
Indeed.

**Comment by ghost** [comment]
Well, it just so happens #10352 was refused

Let's examine this one. It's a system PR, so logically it shouldn't draw gameplay controversy

File: code/game/gamemodes/meteor/meteor.dm [hunk 1]
```diff
@@ -29,62 +24,110 @@
...
+/datum/game_mode/meteor/post_setup()
+
+	//Let's set up the announcement and meteor delay immediatly to send to the admins and use later
+	meteor_announce_delay = rand((meteor_announce_delay_l/600), (meteor_announce_delay_h/600)) * 600 //Minute interval for simplicity
+
+	spawn(300) //Give everything 30 seconds to initialize, this does not delay the rest of post_setup() nor the game and ensures deadmins aren't aware in advance and the admins are
+		message_admins("Meteor storm confirmed by Space Weather Incorporated. Announcement arrives in approximately [round((meteor_announce_delay-200)/600)] minutes, further information will be given then.")
+
+	spawn(rand(waittime_l, waittime_h))
+		if(!mixed) send_intercept()
```
Comment (ComicIronic) [inline code/game/gamemodes/meteor/meteor.dm]: No one-line ifs

File: code/game/gamemodes/meteor/meteor.dm [hunk 2]
```diff
@@ -29,62 +24,110 @@
...
+	world << sound('sound/machines/warning.ogg') //The same chime as the Delta countdown, just twice
 
-	spawn(rand(waittime_l, waittime_h))
-		if(!mixed) send_intercept()
+	if(!meteor_delay)
+		meteor_delay = rand((meteor_delay_l/600), (meteor_delay_h/600))*600 //Let's set up the meteor delay in here
+
+	sleep(20) //Two seconds for warning to play
+
+	if(prob(70)) //Slighty off-scale
```
Comment (ComicIronic) [inline code/game/gamemodes/meteor/meteor.dm:98]: "There should be no gameplay changes"

File: code/game/gamemodes/meteor/meteor.dm (hunk 1 above)
Comment (ghost) [inline code/game/gamemodes/meteor/meteor.dm]: Indeed, give me a moment

File: code/game/gamemodes/meteor/meteor.dm (hunk 2 above)
Comment (ghost) [inline code/game/gamemodes/meteor/meteor.dm:98]: The probability to give a bad estimate for meteor ETA was always in here

**Comment by clusterfack** [comment]
It's on your head if this breaks

**Pull request merged**